### DISCUSSION TITLE
Switch to more verbose file extensions

### DISCUSSION THREAD

**Discussion by lilyremigia** [ppy/osu#14232] [Ideas] [answered]
For shareable third-party content such as beatmapsets, skins, replays, rulesets, a switch or an alternative could be made for more verbose file extensions, since in 2021 we are no longer limited to three-letter ones.
My recommendations is just an osu- prefix, and we could avoid collisions with other software as well, if there was any:
- Beatmapset: `.osu-bms`, since I think we can agree nobody would want to type `.osu-beatmapset`...
Though dropping "beat" could make it decently short enough: `.osu-mapset`...
- Skin: `.osu-skin`
- Replay: `.osu-replay`
- Ruleset?: `.osu-ruleset`
- Collection?: `.osu-collection`

**Comment by bdach** [comment]
"beatmapset" is not a publically-exposed term. It's strictly internal.

While it's true that extensions aren't hard-locked to 3 letters now the suggestions above seem overly verbose.

**Reply by lilyremigia** [reply]
What about `.osudiffs`, `.osuskin`, `.osureplay`, `.osuruleset`, `.osucollection`, `.osubeatmap`, `.osudesign`?

**Comment by peppy** [comment, marked as answer]
i don't understand the goal here. our existing extensions are well defined for what they represent.

we may choose new ones when we move to newer formats. at which point we would consider new naming, but don't need suggestions or discussion for that at this point in time.
